src/app/static/src/js/components/runWorkflow/runWorkflowProgress.vue

```diff
@@ -196,6 +261,39 @@ export default Vue.extend<Data, Methods, unknown, Props>({
                         "An error occurred fetching workflow details";
                 });
         },
+        getWorkflowMetadata(){
+            api.get(`/workflows/${this.selectedWorkflowKey}/`)
+                .then(({data}) => {
+                    this.workflowMetadata = workflowRunDetailsToMetadata(data.data)
+                    this.getReportDependencies()
+                })
+                .catch((error) => {
+                    this.error = error;
+                    this.defaultMessage =
+                        "An error occurred fetching workflow details";
+                });
+        },
+        getReportDependencies() {
+            const commit = this.workflowMetadata.git_commit ? `?commit=${this.workflowMetadata.git_commit}` : '';
+            api.post(`/workflows/summary/${commit}`, {
+                reports: this.workflowMetadata.reports,
+                ref: this.workflowMetadata.git_commit
+            })
+                .then(({data}) => {
+                    const reports = [...this.workflowRunStatus.reports]
+                    this.workflowRunStatus.reports.forEach((report1, index) => {
+                        const extraData = data.data.reports.find(report2 => report2.name === report1.name)
+                        reports[index] = extraData ? {...report1, ...extraData} : report1
+                    });
+                    this.workflowRunStatus = {...this.workflowRunStatus, reports}
```

Comment:
The computed method `anyParams` was not picking up changes to the nested reports hence the heavy use of spread operators. I'd welcome any input on how this could be refactored.

Reply:
This is very confusing, I think this will become simpler once you change the `Type`.

Reply:
Yeah, I'd rather not mutate the `workflowRunStatus`, I think that makes reasoning about the data in the component quite confusing. Couldn't we just keep the summary response in the data too and trust that the indexes of the reports should line up - so in the template, when rendering the parameters, use the `index` of the report in `workflowRunStatus` to access the correct report in `workflowReportParameters`  (or whatever you call the new data property).